packages/apps/plugins/plugin-tree/src/components/Tree/Tree.tsx

```diff
@@ -0,0 +1,321 @@
+        break;
+      case 'Tab':
+        event.preventDefault();
+        onIndent?.(event.shiftKey);
+        break;
+      case 'Enter':
+        // TODO(burdon): Split line (shift to create new).
+        onEnter?.(!!item.text?.length && inputRef.current?.selectionStart === 0);
+        break;
+      case 'Backspace':
+        // TODO(burdon): Merge with previous if caret at start.
+        if (!item.text?.length) {
+          event.preventDefault();
+          onDelete?.();
+        }
+        break;
+      case 'Escape':
+        // TODO(burdon): Toggle checkbox?
+        break;
+    }
+  };
+
+  return (
+    <div className='flex items-center px-2 gap-3'>
+      {(checkbox && (
+        <Input.Root>
+          <Input.Checkbox
+            checked={item.done}
+            onCheckedChange={(checked) => {
+              item.done = !!checked;
+            }}
+          />
+        </Input.Root>
+      )) || (
+        <div className='shrink-0'>
+          <Square weight={active ? 'fill' : undefined} className={mx(getSize(2), active && 'text-primary-500')} />
+        </div>
+      )}
+      <Input.Root>
+        <Input.TextInput
+          ref={inputRef}
+          // autoFocus={active}
+          autoComplete='off'
+          spellCheck={spellCheck}
+          placeholder={focused ? placeholder : undefined}
+          classNames='w-full'
+          variant='subdued'
+          value={item.text ?? ''}
+          onFocus={() => {
+            setFocused(true);
+            onFocus?.();
+          }}
+          onBlur={() => setFocused(false)}
+          onKeyDown={handleKeyDown}
+          onChange={({ target: { value } }) => {
+            item.text = value;
+          }}
+        />
+      </Input.Root>
+      {active && (
+        <DropdownMenu.Root>
+          <DropdownMenu.Trigger asChild>
+            <Button variant='ghost'>
+              <DotsThreeVertical />
+            </Button>
+          </DropdownMenu.Trigger>
+          <DropdownMenu.Portal>
+            <DropdownMenu.Content>
+              <DropdownMenu.Viewport>
+                {onDelete && (
+                  <DropdownMenu.Item onClick={onDelete}>
+                    <X className={getSize(5)} />
+                    <p>Delete item</p>
+                  </DropdownMenu.Item>
+                )}
+              </DropdownMenu.Viewport>
+            </DropdownMenu.Content>
+          </DropdownMenu.Portal>
+        </DropdownMenu.Root>
+      )}
+    </div>
+  );
+};
+
+//
+// Branch
+//
+
+type BranchProps = TreeOptions & {
+  root: Item;
+  active?: string;
+  onFocus?: (item: Item) => void;
+  onCreate?: (parent: Item, item: Item, after?: boolean) => Item;
+  onDelete?: (parent: Item, item: Item) => void;
+  onIndent?: (parent: Item, item: Item, left?: boolean) => void;
+} & Pick<ItemProps, 'onNav'>;
+
+const Branch = ({ root, active, onFocus, onCreate, onDelete, onIndent, onNav, ...props }: BranchProps) => {
+  return (
+    <div className='flex flex-col w-full'>
```

Comment:
Using `flex flex-col` here is probably why you have to specify `is-full` on its children — I think it’s quite likely a `block` layout would require less intervention further down.

Reply:
Great. Deleted both.